pypicloud/cache/dynamo.py

```diff
@@ -22,6 +26,86 @@
 LOG = logging.getLogger(__name__)
 
 
+def _decide_between_versions(contender, current, allow_overwrite):
+    """Decide between packages with the same filename on different paths."""
+
+    if contender.last_modified > current.last_modified:
+        if allow_overwrite:
+            return contender
+    elif contender.last_modified < current.last_modified:
+        if not allow_overwrite:
+            return contender
+    else:
+        def path_depth(package):
+            """Count the slashes in path, or return infinity on no path."""
+            path = package.data.get('path')
+            if not path:
+                return float('+inf')
+            return path.count('/')
```

Comment:
That doesn't cover the case where you have two files with the same filename (but different paths) on storage. This methods basically encapsulates which of the two the rebuild prefers. The cache will have the path of the chosen file.